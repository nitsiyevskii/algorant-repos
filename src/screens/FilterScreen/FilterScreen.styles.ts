import { createStyles } from 'src/utils/createStyles';

export const useStyles = createStyles((theme) => ({
  container: {
    flex: 1,
    backgroundColor: theme.colors.background.primary,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: theme.colors.separator.primary,
  },
  closeButton: {
    padding: 4,
  },
  headerTitle: {
    fontSize: theme.fonts.size.xl,
    fontWeight: theme.fonts.weight.semibold,
    color: theme.colors.text.primary,
  },
  resetButton: {
    fontSize: theme.fonts.size.md,
    color: theme.colors.core.primary,
    fontWeight: theme.fonts.weight.semibold,
  },
  content: {
    flex: 1,
  },
  section: {
    paddingHorizontal: 16,
    paddingVertical: 20,
    borderBottomWidth: 1,
    borderBottomColor: theme.colors.separator.primary,
  },
  footer: {
    padding: 16,
    borderTopWidth: 1,
    borderTopColor: theme.colors.separator.primary,
    backgroundColor: theme.colors.background.primary,
  },
  applyButton: {
    backgroundColor: theme.colors.core.primary,
    borderRadius: 8,
    paddingVertical: 14,
    alignItems: 'center',
  },
  applyButtonText: {
    fontSize: theme.fonts.size.md,
    fontWeight: theme.fonts.weight.semibold,
    color: theme.colors.button.text,
  },
}));

