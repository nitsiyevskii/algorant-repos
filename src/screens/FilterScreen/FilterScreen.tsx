import React from "react";
import { View, Text, ScrollView, TouchableOpacity } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { router } from "expo-router";
import { Ionicons } from "@expo/vector-icons";
import { useTheme } from "src/hooks/useTheme";
import { ORGANIZATION_OPTIONS, AVAILABLE_LANGUAGES } from "src/constants";
import { useAppDispatch, useAppSelector } from "src/store/hooks";
import {
  toggleOrganization,
  toggleLanguage,
  setStarMin,
  setStarMax,
  setForkMin,
  setForkMax,
  resetFilters,
} from "src/store/slices/filtersSlice";
import {
  selectOrganizationFilters,
  selectLanguageFilters,
  selectStarFilters,
  selectForkFilters,
} from "src/store/selectors/repositoriesSelectors";
import { MultiSelectField } from "./components/MultiSelectField/MultiSelectField";
import { RangeInput } from "./components/RangeInput/RangeInput";
import { useStyles } from "./FilterScreen.styles";

export const FilterScreen = () => {
  const theme = useTheme();
  const styles = useStyles();
  const dispatch = useAppDispatch();

  const selectedOrganizations = useAppSelector(selectOrganizationFilters);
  const selectedLanguages = useAppSelector(selectLanguageFilters);
  const stars = useAppSelector(selectStarFilters);
  const forks = useAppSelector(selectForkFilters);

  const handleToggleOrganization = (org: string) => {
    dispatch(toggleOrganization(org));
  };

  const handleToggleLanguage = (lang: string) => {
    dispatch(toggleLanguage(lang));
  };

  const handleStarMinChange = (value: number | null) => {
    dispatch(setStarMin(value));
  };

  const handleStarMaxChange = (value: number | null) => {
    dispatch(setStarMax(value));
  };

  const handleForkMinChange = (value: number | null) => {
    dispatch(setForkMin(value));
  };

  const handleForkMaxChange = (value: number | null) => {
    dispatch(setForkMax(value));
  };

  const handleReset = () => {
    dispatch(resetFilters());
  };

  const handleApply = () => {
    router.back();
  };

  return (
    <SafeAreaView style={styles.container} edges={["bottom"]}>
      <View style={styles.header}>
        <TouchableOpacity
          onPress={() => router.back()}
          style={styles.closeButton}
        >
          <Ionicons name="close" size={28} color={theme.colors.text.primary} />
        </TouchableOpacity>
        <Text style={styles.headerTitle}>Filters</Text>
        <TouchableOpacity onPress={handleReset}>
          <Text style={styles.resetButton}>Reset</Text>
        </TouchableOpacity>
      </View>

      <ScrollView style={styles.content} showsVerticalScrollIndicator={false}>
        <View style={styles.section}>
          <RangeInput
            label="Stars"
            minValue={stars.min}
            maxValue={stars.max}
            onMinChange={handleStarMinChange}
            onMaxChange={handleStarMaxChange}
          />
        </View>
        <View style={styles.section}>
          <RangeInput
            label="Forks"
            minValue={forks.min}
            maxValue={forks.max}
            onMinChange={handleForkMinChange}
            onMaxChange={handleForkMaxChange}
          />
        </View>
        <View style={styles.section}>
          <MultiSelectField
            label="Organization"
            options={ORGANIZATION_OPTIONS}
            selectedValues={selectedOrganizations}
            onToggle={handleToggleOrganization}
            type="checkbox"
          />
        </View>
        <View style={styles.section}>
          <MultiSelectField
            label="Language"
            options={AVAILABLE_LANGUAGES}
            selectedValues={selectedLanguages}
            onToggle={handleToggleLanguage}
            type="chip"
          />
        </View>
      </ScrollView>

      <View style={styles.footer}>
        <TouchableOpacity style={styles.applyButton} onPress={handleApply}>
          <Text style={styles.applyButtonText}>Apply Filters</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};
