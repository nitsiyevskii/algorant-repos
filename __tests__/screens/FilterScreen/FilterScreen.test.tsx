import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { useDispatch, useSelector } from 'react-redux';
import { router } from 'expo-router';
import { FilterScreen } from 'src/screens/FilterScreen/FilterScreen';

jest.mock('react-redux');

const mockDispatch = jest.fn();
const mockUseDispatch = useDispatch as unknown as jest.Mock;
const mockUseSelector = useSelector as unknown as jest.Mock;

describe('FilterScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockUseDispatch.mockReturnValue(mockDispatch);
    
    mockUseSelector.mockImplementation((selector) => {
      const state = {
        filters: {
          organizations: [],
          languages: [],
          stars: { min: null, max: null },
          forks: { min: null, max: null },
          searchQuery: '',
        },
      };
      return selector(state);
    });
  });

  it('renders correctly', () => {
    const { getByText } = render(<FilterScreen />);

    expect(getByText('Filters')).toBeTruthy();
    expect(getByText('Reset')).toBeTruthy();
    expect(getByText('Apply Filters')).toBeTruthy();
  });

  it('renders all filter sections', () => {
    const { getByText } = render(<FilterScreen />);

    expect(getByText('Organization')).toBeTruthy();
    expect(getByText('Language')).toBeTruthy();
    expect(getByText('Stars')).toBeTruthy();
    expect(getByText('Forks')).toBeTruthy();
  });

  it('renders organization options', () => {
    const { getByText } = render(<FilterScreen />);

    expect(getByText('Pera Wallet')).toBeTruthy();
    expect(getByText('Algorand Foundation')).toBeTruthy();
    expect(getByText('Algorand')).toBeTruthy();
  });

  it('renders language options', () => {
    const { getByText } = render(<FilterScreen />);

    expect(getByText('TypeScript')).toBeTruthy();
    expect(getByText('JavaScript')).toBeTruthy();
    expect(getByText('Python')).toBeTruthy();
    expect(getByText('Go')).toBeTruthy();
  });

  it('calls router.back when close button is pressed', () => {
    const { getByTestId } = render(<FilterScreen />);

    const closeButton = getByTestId('icon');
    fireEvent.press(closeButton.parent);

    expect(router.back).toHaveBeenCalledTimes(1);
  });

  it('calls router.back when apply button is pressed', () => {
    const { getByText } = render(<FilterScreen />);

    const applyButton = getByText('Apply Filters');
    fireEvent.press(applyButton);

    expect(router.back).toHaveBeenCalledTimes(1);
  });

  it('dispatches toggleOrganization when organization is toggled', () => {
    const { getByTestId } = render(<FilterScreen />);

    const checkbox = getByTestId('checkbox-perawallet');
    fireEvent.press(checkbox);

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('dispatches toggleLanguage when language is toggled', () => {
    const { getByTestId } = render(<FilterScreen />);

    const chip = getByTestId('chip-TypeScript');
    fireEvent.press(chip);

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('dispatches resetFilters when reset button is pressed', () => {
    const { getByText } = render(<FilterScreen />);

    const resetButton = getByText('Reset');
    fireEvent.press(resetButton);

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('dispatches setStarMin when star min input changes', () => {
    const { getByTestId } = render(<FilterScreen />);

    const minInput = getByTestId('range-input-min-stars');
    fireEvent.changeText(minInput, '100');

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('dispatches setStarMax when star max input changes', () => {
    const { getByTestId } = render(<FilterScreen />);

    const maxInput = getByTestId('range-input-max-stars');
    fireEvent.changeText(maxInput, '500');

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('dispatches setForkMin when fork min input changes', () => {
    const { getByTestId } = render(<FilterScreen />);

    const minInput = getByTestId('range-input-min-forks');
    fireEvent.changeText(minInput, '10');

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('dispatches setForkMax when fork max input changes', () => {
    const { getByTestId } = render(<FilterScreen />);

    const maxInput = getByTestId('range-input-max-forks');
    fireEvent.changeText(maxInput, '50');

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('displays selected filters from Redux state', () => {
    mockUseSelector.mockImplementation((selector) => {
      const state = {
        filters: {
          organizations: ['perawallet'],
          languages: ['TypeScript', 'JavaScript'],
          stars: { min: 100, max: 500 },
          forks: { min: 10, max: 50 },
          searchQuery: '',
        },
      };
      return selector(state);
    });

    const { getByDisplayValue } = render(<FilterScreen />);

    expect(getByDisplayValue('100')).toBeTruthy();
    expect(getByDisplayValue('500')).toBeTruthy();
    expect(getByDisplayValue('10')).toBeTruthy();
    expect(getByDisplayValue('50')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<FilterScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
