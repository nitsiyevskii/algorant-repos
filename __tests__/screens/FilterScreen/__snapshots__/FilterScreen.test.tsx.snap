// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterScreen matches snapshot 1`] = `
<View
  edges={
    [
      "bottom",
    ]
  }
  style={
    {
      "backgroundColor": "#F2F2F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#C6C6C8",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
    >
      <Text
        color="#000000"
        name="close"
        size={28}
        testID="icon"
      >
        close
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 20,
          "fontWeight": "600",
        }
      }
    >
      Filters
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#007AFF",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Reset
      </Text>
    </View>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "borderBottomColor": "#C6C6C8",
            "borderBottomWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            {
              "marginBottom": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "600",
                "marginBottom": 16,
              }
            }
          >
            Stars
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#3C3C43",
                    "fontSize": 14,
                    "marginBottom": 8,
                  }
                }
              >
                Min
              </Text>
              <TextInput
                keyboardType="number-pad"
                onChangeText={[Function]}
                placeholder="0"
                placeholderTextColor="#8E8E93"
                style={
                  {
                    "backgroundColor": "#F2F2F7",
                    "borderColor": "#C7C7CC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 16,
                    "paddingHorizontal": 12,
                    "paddingVertical": 10,
                  }
                }
                testID="range-input-min-stars"
                value=""
              />
            </View>
            <Text
              style={
                {
                  "color": "#3C3C43",
                  "fontSize": 18,
                  "marginHorizontal": 12,
                  "marginTop": 20,
                }
              }
            >
              -
            </Text>
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#3C3C43",
                    "fontSize": 14,
                    "marginBottom": 8,
                  }
                }
              >
                Max
              </Text>
              <TextInput
                keyboardType="number-pad"
                onChangeText={[Function]}
                placeholder="∞"
                placeholderTextColor="#8E8E93"
                style={
                  {
                    "backgroundColor": "#F2F2F7",
                    "borderColor": "#C7C7CC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 16,
                    "paddingHorizontal": 12,
                    "paddingVertical": 10,
                  }
                }
                testID="range-input-max-stars"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#C6C6C8",
            "borderBottomWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            {
              "marginBottom": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "600",
                "marginBottom": 16,
              }
            }
          >
            Forks
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#3C3C43",
                    "fontSize": 14,
                    "marginBottom": 8,
                  }
                }
              >
                Min
              </Text>
              <TextInput
                keyboardType="number-pad"
                onChangeText={[Function]}
                placeholder="0"
                placeholderTextColor="#8E8E93"
                style={
                  {
                    "backgroundColor": "#F2F2F7",
                    "borderColor": "#C7C7CC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 16,
                    "paddingHorizontal": 12,
                    "paddingVertical": 10,
                  }
                }
                testID="range-input-min-forks"
                value=""
              />
            </View>
            <Text
              style={
                {
                  "color": "#3C3C43",
                  "fontSize": 18,
                  "marginHorizontal": 12,
                  "marginTop": 20,
                }
              }
            >
              -
            </Text>
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#3C3C43",
                    "fontSize": 14,
                    "marginBottom": 8,
                  }
                }
              >
                Max
              </Text>
              <TextInput
                keyboardType="number-pad"
                onChangeText={[Function]}
                placeholder="∞"
                placeholderTextColor="#8E8E93"
                style={
                  {
                    "backgroundColor": "#F2F2F7",
                    "borderColor": "#C7C7CC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 16,
                    "paddingHorizontal": 12,
                    "paddingVertical": 10,
                  }
                }
                testID="range-input-max-forks"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#C6C6C8",
            "borderBottomWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            {
              "marginBottom": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "600",
                "marginBottom": 16,
              }
            }
          >
            Organization
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingVertical": 12,
              }
            }
            testID="checkbox-perawallet"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#007AFF",
                  "borderRadius": 6,
                  "borderWidth": 2,
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 12,
                  "width": 24,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 16,
                }
              }
            >
              Pera Wallet
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingVertical": 12,
              }
            }
            testID="checkbox-algorandfoundation"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#007AFF",
                  "borderRadius": 6,
                  "borderWidth": 2,
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 12,
                  "width": 24,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 16,
                }
              }
            >
              Algorand Foundation
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingVertical": 12,
              }
            }
            testID="checkbox-algorand"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#007AFF",
                  "borderRadius": 6,
                  "borderWidth": 2,
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 12,
                  "width": 24,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 16,
                }
              }
            >
              Algorand
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#C6C6C8",
            "borderBottomWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            {
              "marginBottom": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "600",
                "marginBottom": 16,
              }
            }
          >
            Language
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "gap": 8,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-TypeScript"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                TypeScript
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-JavaScript"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                JavaScript
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-Python"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                Python
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-Go"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                Go
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-Java"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                Java
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-Rust"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                Rust
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-Swift"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                Swift
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#E5E5EA",
                  "borderColor": "#C6C6C8",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
              testID="chip-Kotlin"
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    false,
                  ]
                }
              >
                Kotlin
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "backgroundColor": "#F2F2F7",
        "borderTopColor": "#C6C6C8",
        "borderTopWidth": 1,
        "padding": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#007AFF",
          "borderRadius": 8,
          "opacity": 1,
          "paddingVertical": 14,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Apply Filters
      </Text>
    </View>
  </View>
</View>
`;
